parallel-install/pkg/deployment/overrideinterceptors.go

```diff
@@ -381,3 +381,57 @@ func defaultRetryOptions() []retry.Option {
 		retry.DelayType(retry.FixedDelay),
 	}
 }
+
+type RegistryInterceptor struct {
+	kubeClient                kubernetes.Interface
+	enableInternalRegistryKey string
+	serverAddressKey          string
+	internalServerAddressKey  string
+	registryAddressKey        string
+}
+
+func NewReigstryInterceptor(kubeClient kubernetes.Interface, enableInternalRegistryKey string, serverAddressKey string, internalServerAddressKey string, registryAddressKey string) *RegistryInterceptor {
+	return &RegistryInterceptor{
+		kubeClient:                kubeClient,
+		enableInternalRegistryKey: enableInternalRegistryKey,
+		serverAddressKey:          serverAddressKey,
+		internalServerAddressKey:  internalServerAddressKey,
+		registryAddressKey:        registryAddressKey,
+	}
+}
+
+func (r RegistryInterceptor) String(value interface{}, key string) string {
+	return fmt.Sprintf("%v", value)
```

Comment:
Shouldn't this return the intercepted value instead of the default/user defined in case of k3s?